libcontainer/container_linux.go

```diff
@@ -523,10 +544,32 @@ func (c *linuxContainer) newInitProcess(p *Process, cmd *exec.Cmd, messageSockPa
 		}
 	}
 	_, sharePidns := nsMaps[configs.NEWPID]
-	data, err := c.bootstrapData(c.config.Namespaces.CloneFlags(), nsMaps)
+	data, err := c.bootstrapData(c.config.Namespaces.CloneFlags(), nsMaps, initStandard)
 	if err != nil {
 		return nil, err
 	}
+
+	if c.shouldSendMountSources() {
+		mountFileFdsList := ""
+		for _, m := range c.config.Mounts {
+			if m.Device != "bind" {
+				// StartInitialization() finds out the Mounts indices by counting ";".
+				// We take care of adding the right number of ";"
+				mountFileFdsList += ";"
+				continue
+			}
+			// The fd passed here will not be used: nsexec.c will overwrite it with dup3(). We just need
+			// to allocate a fd so that we know the number to pass in the environment variable. The fd
+			// must not be closed before cmd.Start(), so we reuse messageSockPair.child because the
+			// lifecycle of that fd is already taken care of.
+			cmd.ExtraFiles = append(cmd.ExtraFiles, messageSockPair.child)
+			mountFileFdsList += fmt.Sprintf("%d;", stdioFdCount+len(cmd.ExtraFiles)-1)
```

Comment:
I realize it's a micro-optimization, but can we use `strings.Builder` and `strconv.Itoa` here instead? Go strings are immutable, every time we append to a string a new one is created.

I think using `strings.Builder` will result in less garbage to collect. `fmt.Sprintf` is also much slower than `strconv.Itoa`.

Reply:
Sure, will check how to use that.

btw, the `append()` is being used in several other places, also in the `_LIBCONTAINER_` vars. Should I create an issue to replace them, at least in those vars?

Reply:
It's not about `append()`, it's about appending to a string, i.e. `str += ...` (or other similar string manipulation).

Reply:
Cool, switched to `strings.Builder`.

I used `Grow()` with a likely guess for the top, to reduce (re)allocations in most cases, but I'm not sure it is worth trying to guess. Let me know if you think it is better to remove that line, my personal taste is to remove it, but added it to be in-line with reducing allocations, given that we can do educated guess quite easily here.